huaweicloud/services/sfs/resource_huaweicloud_sfs_turbo_dir.go

```diff
@@ -0,0 +1,185 @@
+package sfs
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func ResourceSfsTurboDir() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceSfsTurboDirCreate,
+		ReadContext:   resourceSfsTurboDirRead,
+		DeleteContext: resourceSfsTurboDirDelete,
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+			"share_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the SFS Turbo ID.`,
+			},
+			"path": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the valid full path of SFS Turbo directory.`,
+			},
+			"mode": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				ForceNew:    true,
+				Description: `Specifies the SFS Turbo directory permissions.`,
+			},
+			"uid": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				ForceNew:    true,
+				Description: `Specifies the user ID of the SFS Turbo directory.`,
+			},
+			"gid": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				ForceNew:    true,
+				Description: `Specifies the group ID of the SFS Turbo directory.`,
+			},
+		},
+	}
+}
+
+func resourceSfsTurboDirCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg     = meta.(*config.Config)
+		region  = cfg.GetRegion(d)
+		httpUrl = "v1/{project_id}/sfs-turbo/shares/{share_id}/fs/dir"
+	)
+	sfsClient, err := cfg.SfsV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating SFS v1 client: %s", err)
+	}
+
+	createPath := sfsClient.Endpoint + httpUrl
+	createPath = strings.ReplaceAll(createPath, "{project_id}", sfsClient.ProjectID)
+	createPath = strings.ReplaceAll(createPath, "{share_id}", d.Get("share_id").(string))
+	createOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody:         buildCreateSfsTurboDirBodyParams(d),
```

Comment:
it's better to use `utils.RemoveNil` to warp it, then it's unnecessary to use d.GetOk in buildCreateSfsTurboDirBodyParams